### ISSUE TITLE
Include open-vm-tools

### ISSUE THREAD

**Issue by patricvisit** [rancher/os#219]
Please include open-vm-tools for better compatibility with ESXi and VMWare workstation.

https://github.com/vmware/open-vm-tools
http://sourceforge.net/projects/open-vm-tools/files/open-vm-tools/

**Comment by ibuildthecloud** [comment]
Will do.  Just curious though.  What features are you looking to get out of these tools?  Operations around mounts are tricky at the moment.

**Comment by patricvisit** [comment]
Well from the FAQ, the things that would be nice to have are:
- Memory ballooning
- Disk wiping and shrinking (maybe)
- Time synchronization (maybe, though NTP is a better choice)
- GuestInfo (provides statistics about guest environment) <- very useful
- Soft power operations <- very useful

http://open-vm-tools.sourceforge.net/faq.php

**Comment by ibuildthecloud** [comment]
@imikushin Can you check if buildroot already has open-vm-tools?

**Comment by imikushin** [comment]
@ibuildthecloud Looks like it does: https://github.com/buildroot/buildroot/tree/master/package/openvmtools

**Comment by kbudde** [comment]
Hi,

i'm deploying vms to vmware esxi with docker-machine. I've tried to do it with rancheros as well:

    docker-machine create -d vmwarevsphere --vmwarevsphere-boot2docker-url https://releases.rancher.com/os/latest/rancheros.iso myVM

VM is created. BUT vmware tools are not running. docker-machine is waiting for ip from guestinfo. As the tools are not running the machine is not successfully deployed :(

**Comment by dsvmacdonald** [comment]
Same problem for vmware fusion.  Cannot create a docker machine using RancherOs and vmwarefusion driver.  Any idea on when vmware fusion will be supported?

**Comment by sysgen** [comment]
Hey guys, you really must include vm tools support out of the box.  Without it, its not considered 'enterprise grade' and just wont get a foothold in the many corporates with virtualized infrastructure.   Any update on when we can expect support?
Thanks

**Comment by mlham** [comment]
I am in the same boat. We use VMWare at almost all the clients I work with ... I can pretty much promise you that virtualbox isn't an option in any of them ... well maybe for like five developers ... but that is it.

If they can't see it in the standard tooling ... then they simply will not use it. Right now we are just getting their feet wet with standard "docker" in a guest (RHEL6 and RHEL7) ... but for sure their will be no traction for this if there aren't the hooks for them to use/see it in their standard process ... which is VMWare based.

**Comment by ibuildthecloud** [comment]
We will for sure address this in 0.5.  We do have users on VMware today that just pull in the VM tools themselves but the fact that it isn't there by default does give the impression we don't care about VMware (which is not the case)

**Comment by ikappas** [comment]
Hey guys, is there a guide on how to pull in the VM tools for 0.4.4?

**Comment by liquidstate** [comment]
+1  Any ETA for this?

**Comment by joshwget** [comment]
Confirmed with rancher/os-v0.5.0-rc1. Soft power operations are still a known issue, so I'll file a separate issue regarding those. If anyone finds further pieces of functionality that are missing, please file individual issues for them!

**Comment by 2mind** [comment]
Nice. But i catch a bug:
https://bugs.alpinelinux.org/issues/5105

[root@vm-devops-docker-template ~]# cat /etc/os-release 
NAME="RancherOS"
VERSION=v0.5.0
ID=rancheros
ID_LIKE=busybox
VERSION_ID=v0.5.0
PRETTY_NAME="RancherOS"
HOME_URL=
SUPPORT_URL=
BUG_REPORT_URL=
BUILD_ID=
[root@vm-devops-docker-template ~]# system-docker run -it rancher/os-openvmtools:v0.5.0-3 sh
/ # cat /etc/issue 
Welcome to Alpine Linux 3.4
Kernel \r on an \m (\l)

**Comment by chinaares** [comment]
0.5 not work for me!
still blocked by the message:
Waiting for VMware Tools to come online...

**Comment by fighne** [comment]
So what is the current status for this as it's very confusing at the moment?

**Comment by PhilipSchmid** [comment]
Hi there,

It seems to me, like the tools are now included. See https://docs.rancher.com/os/system-services/adding-system-services/
```
sudo ros service list
sudo ros service enable open-vm-tools
Pulling open-vm-tools (rancher/os-openvmtools:v0.7.1)...
v0.7.1: Pulling from rancher/os-openvmtools
952132ac251a: Pull complete 
82659f8f1b76: Pull complete 
c19118ca682d: Pull complete 
8296858250fe: Pull complete 
24e0251a0e2c: Pull complete 
3c5b40f36180: Pull complete 
Digest: sha256:3a19dd5db8ff16cfd7f3ceed74eccb33591112d15385eb4820b580ec0b7a27be
Status: Downloaded newer image for rancher/os-openvmtools:v0.7.1
sudo ros service start open-vm-tools 
```
Unfortunately the vCenter still shows "Not running, not installed". Any guesses what I missed?

Regards,
Philip

**Comment by gizmotronic** [comment]
@PhilipSchmid, perhaps you could add some details of your environment. I can say that it worked for me using vSphere 5.5u3 (standalone) when I installed the OS to the guest's disk, with both the default console and the Ubuntu console. When booted using iPXE, it worked with the default console but not the Ubuntu console. When it's working it shows as "Running (3rd-party/Independent)".

**Comment by PhilipSchmid** [comment]
Hi @gizmotronic,

I use RancherOS v0.7.1 on a VMware vSphere 6.0 environment.

```bash
[rancher@node1 ~]$ sudo ros os version
v0.7.1
[rancher@node1 ~]$ sudo ros service list
disabled amazon-ecs-agent
disabled kernel-extras
disabled kernel-headers
disabled kernel-headers-system-docker
enabled  open-vm-tools
```

What kind of additional information would you like to have?

Perhaps off topic, but I find it suspicious: The ros v0.7.1 Image is not local...
```bash
[rancher@inslnxcl01 ~]$ sudo ros os list
rancher/os:v0.4.0 remote
rancher/os:v0.4.1 remote
rancher/os:v0.4.2 remote
rancher/os:v0.4.3 remote
rancher/os:v0.4.4 remote
rancher/os:v0.4.5 remote
rancher/os:v0.5.0 remote
rancher/os:v0.6.0 remote
rancher/os:v0.6.1 remote
rancher/os:v0.7.0 remote
rancher/os:v0.7.1 remote
```

Regards,
Philip

**Comment by gizmotronic** [comment]
@PhilipSchmid Glad you figured it out. I had an opportunity to go back and test this on a build machine I was converting to Rancher OS today. Enabling open-vm-tools didn't start the service, so I tried starting it manually. That also didn't do anything. As in your case, though, it came right up on a subsequent reboot.
